test/common/services/kibana_server/extend_es_archiver.js

```diff
@@ -25,7 +25,7 @@ export function extendEsArchiver({ esArchiver, kibanaServer, retry, defaults })
       const statsKeys = Object.keys(stats);
       const kibanaKeys = statsKeys.filter(
         // this also matches stats keys like '.kibana_1' and '.kibana_2,.kibana_1'
-        (key) => key.includes(KIBANA_INDEX) && (stats[key].created || stats[key].deleted)
+        (key) => key.includes(KIBANA_INDEX) && stats[key].created
```

Comment:
there's no point in restoring uiSettings defaults when we delete a saved objects index since tests will have to explicitly initialize a new index anyway (at which point we will restore the defaults)